lib/Drupal/service_container/Tests/LegacyDrupal7Test.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\service_container\Tests\LegacyDrupal7Test.
+ */
+
+namespace Drupal\service_container\Tests;
+
+use Drupal\service_container\Legacy\Drupal7;
+
+class LegacyDrupal7Test extends ServiceContainerIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getInfo() {
+    return array(
+      'name' => 'LegacyDrupal7Test',
+      'description' => 'Tests the legacy Drupal 7 integration service',
+      'group' => 'service_container',
+    );
+  }
+
+  public function testLegacyDrupal7() {
+    /** @var \Drupal\service_container\Legacy\Drupal7 $drupal7_service */
+
+    $drupal7_service = $this->container->get('drupal7');
+    $this->assertTrue($drupal7_service instanceof Drupal7);
+
+    $random_message = $this->randomString();
+    drupal_set_message($random_message, 'warning');
+    $messages = $drupal7_service->drupal_get_messages();
```

Comment:
Final review:

Could we switch those two?

$drupal7_service->drupal_set_message($random_message, 'warning');
$messages = drupal_get_messages();

I would like to ensure we test with arguments.